### DISCUSSION TITLE
TypeError when updating float values on form with empty input

### DISCUSSION THREAD

**Discussion by lsmedia** [livewire/livewire#8124] [1. Help]
I use a Livewire component in combination with a Livewire form. The form has a public (nullable) float value. When I empty the input field on the interface, it tries to update the float field with an empty string. This topic has already been discussed elsewhere and it is a desired behaviour that empty strings are transported in this case.

However, this triggers a `TypeError` in combination with float variables on Forms. As soon as I take the variable out of the form and place it directly in the component, everything works without any problems and the float variable takes on the value `null`.

The problem does not occur with integer variables - only when using float variables.

Here is an example and my attempt to solve the problem with the updating hook, which does not seem to work:

The component:

```
use Livewire\Component;

class TestComponent extends Component
{
    public TestForm $form;

    // this variable is filled as expected
    public ?float $floatVal = null;
    

    // my try to fix the issue
    public function updatingFormFloatVal(&$value)
    {
        if ($value == "") {
            $value = null;      // this does not seem to solve the problem
        }
    }
}
```

The form:

```
use Livewire\Form;

class TestForm extends Form
{
    // updating this variable throws a TypeError if input is empty
    public ?float $floatVal = null;
}
```

The view:

```
<div>
    <!-- this input field causes a TypeError as soon as it is emptied -->
    <input wire:model.live="form.floatVal" type="number" step="0.1">

    <!-- This works because the variable is stored directly in the component -->
    <input wire:model.live="floatVal" type="number" step="0.1">
</div>
```

Does anyone have a tip on how I can handle empty strings before updating float values in the form to avoid TypeErrors?

I'm using:
Livewire v3.4.9
Laravel v10.48.2
PHP 8.2.16 (cli)

**Comment by joshhanley** [comment]
@lsmedia we currently have an `IntSynth` for handling integers, we probably need one for floats. PRs welcome!

https://github.com/livewire/livewire/blob/main/src/Mechanisms/HandleComponents/Synthesizers/IntSynth.php

**Reply by lsmedia** [reply]
Thank you very much for drawing my attention to the synthesizers.

 I was able to trace the problem and fix it with the changes in the following PR: https://github.com/livewire/livewire/pull/8126
